### PR TITLE
Fix for SS Login and Logout handlers not being invoked

### LINKED ISSUE #915: StormpathFilter must be behind Spring Security Filter
Relates to #892 #1145

### LINKED ISSUE #1145: Spring Security sends out 2 authentication attempts
While carrying out some troubleshooting in `stormpath-sdk-examples-spring-security-spring-boot-webmvc` I see that 2 authentications attempt are executed (i.e. traveling through the wire to the backend):

1. In `LoginController#onValidSubmit` -> req.login(usernameOrEmail, password);
2. `StormpathAuthenticationProvider#authenticate`

This issue also causes the `PostLoginHandler` to be invoked when the `Authentication` instance is not yet available in the `SecurityContext`. Therefore a post login handler will not be able to see the granted authorities for the logged in account.

### LINKED ISSUE #1158: Spring Security Login and Logout Handlers are not being invoked
I see that `StormpathLoginSuccessHandler` is never invoked. `StormpathLogoutHandler` is also not being invoked during logout.

### PR DESCRIPTION
**Pull request by mrioan**
Resolves #1158 
Resolves #1145 
Resolves #915 

The implemented solution was to move spring security in front of Stormpath

### CODE REVIEW

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java [hunk 1]
```diff
@@ -18,11 +18,17 @@
...
 import com.stormpath.sdk.saml.SamlResultListener;
 import com.stormpath.sdk.servlet.csrf.CsrfTokenManager;
+import com.stormpath.sdk.servlet.filter.account.AccountResolverFilter;
 import com.stormpath.sdk.servlet.mvc.ErrorModelFactory;
 import com.stormpath.spring.config.AbstractStormpathWebSecurityConfiguration;
 import com.stormpath.spring.config.StormpathWebSecurityConfigurer;
-import com.stormpath.spring.filter.SpringSecurityResolvedAccountFilter;
-import com.stormpath.spring.oauth.OAuthAuthenticationSpringSecurityProcessingFilter;
+import com.stormpath.spring.filter.ContentNegotiationSpringSecurityAuthenticationFilter;
+//import com.stormpath.spring.filter.SpringSecurityResolvedAccountFilter;
```
Comment (dogeared) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java]: remove commented out imports

File: extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java [hunk 2]
```diff
@@ -680,4 +691,5 @@ public AccessTokenControllerConfig stormpathAccessTokenConfig() {
     public GrantTypeValidator stormpathGrantTypeStatusValidator() {
         return super.stormpathGrantTypeStatusValidator();
     }
+
```
Comment (dogeared) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java]: nit: don't need the newline here

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java [hunk 3]
```diff
@@ -162,17 +222,84 @@ public CsrfTokenManager stormpathCsrfTokenManager() {
...
+     */
+    public AccountResolverFilter springSecuritResolvedAccountFilter() {
+        AccountResolverFilter accountResolverFilter = new AccountResolverFilter();
+        accountResolverFilter.setEnabled(stormpathFilterEnabled);
+        accountResolverFilter.setResolvers(stormpathAccountResolvers);
+        accountResolverFilter.setOauthEndpointUri(accessTokenUri);
+        return accountResolverFilter;
     }
 
+
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java]: nit: extra newline

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 4]
```diff
@@ -240,17 +276,27 @@ public void init(HttpSecurity http) throws Exception {
         context.getAutowireCapableBeanFactory().autowireBean(this);
         http.servletApi().rolePrefix(""); //Fix for https://github.com/stormpath/stormpath-sdk-java/issues/325
 
-        // We need to add the springSecurityResolvedAccountFilter whenever we have our login enabled in order to
-        // fix https://github.com/stormpath/stormpath-sdk-java/issues/450
-        http.addFilterBefore(springSecurityResolvedAccountFilter, AnonymousAuthenticationFilter.class);
-
         if (loginEnabled) {
+
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: nit: extra newline

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 5]
```diff
@@ -240,17 +276,27 @@ public void init(HttpSecurity http) throws Exception {
...
+            http.addFilterBefore(stormpathWrapperFilter, SecurityContextPersistenceFilter.class);
+
+            // We need to add the springSecurityResolvedAccountFilter whenever we have our login enabled in order to
+            // fix https://github.com/stormpath/stormpath-sdk-java/issues/450
+            http.addFilterBefore(springSecurityResolvedAccountFilter, LogoutFilter.class);
+
+            //Fix for redirection loop when Cookie is present but WebApp is restarted and '/' is locked down to authenticated users (Bare Bones example)
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: nit: above comments have space after //

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 6]
```diff
@@ -240,17 +276,27 @@ public void init(HttpSecurity http) throws Exception {
...
             // This filter replaces http.formLogin() so that we can properly handle content negotiation
             // If it's an HTML request, it delegates to the default UsernamePasswordAuthenticationFilter behavior
             // refer to: https://github.com/stormpath/stormpath-sdk-java/issues/682
-            http.addFilterBefore(setupContentNegotiationAuthenticationFilter(), UsernamePasswordAuthenticationFilter.class);
+            http.addFilterBefore(contentNegotiationSpringSecurityAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
+        }
 
-            http.addFilterBefore(preLoginHandlerFilter(), ContentNegotiationAuthenticationFilter.class);
+        if (corsEnabled) {
+            http.cors(); // Let's add Spring Security's built-in support for CORs
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: nice!

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/LoginHandlerFilter.java [hunk 7]
```diff
@@ -12,7 +27,9 @@
 /**
  * @since 1.0.0
  */
+//TODO: Remove as this filter seems to be un-used
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/LoginHandlerFilter.java]: Why wait to remove if it's unused?

File: extensions/spring/stormpath-spring-security-webmvc/src/test/java/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcTestAppConfig.java [hunk 8]
```diff
@@ -15,11 +15,49 @@
...
  * @since 1.0.RC5
  */
 @Configuration
 @EnableStormpathWebSecurity
-public class MinimalStormpathSpringSecurityWebMvcTestAppConfig {}
+public class MinimalStormpathSpringSecurityWebMvcTestAppConfig  extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.apply(stormpath());
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/test/java/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcTestAppConfig.java]: Isn't this going away? Is it needed in the context of tests?

File: extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java [hunk 9]
```diff
@@ -853,11 +863,11 @@ public HeaderAuthenticator stormpathAuthorizationHeaderAuthenticator() {
 
     public Resolver<Account> stormpathCookieAccountResolver() {
         return new CookieAccountResolver(
-            stormpathAccessTokenCookieConfig(),
-            stormpathRefreshTokenCookieConfig(),
-            stormpathJwtAccountResolver(),
-            stormpathCookieAuthenticationResultSaver(),
-            stormpathAccessTokenResultFactory());
+                stormpathAccessTokenCookieConfig(),
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java]: Why the extra indents?

File: extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebMvcConfiguration.java [hunk 10]
```diff
@@ -532,4 +542,5 @@ public AccessTokenControllerConfig stormpathAccessTokenConfig(){
     public GrantTypeValidator stormpathGrantTypeStatusValidator(){
         return super.stormpathGrantTypeStatusValidator();
     }
+
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebMvcConfiguration.java]: nit: extra newline

File: impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java [hunk 11]
```diff
@@ -399,7 +399,7 @@ private void cache(Class<? extends Resource> clazz, Map<String, ?> data, QuerySt
             return returnValue;
         } catch (Exception e) {
             throw new IllegalStateException(
-                "Unable to access PROPERTY_DESCRIPTORS static field on implementation class " + clazz.getName(), e);
+                    "Unable to access PROPERTY_DESCRIPTORS static field on implementation class " + clazz.getName(), e);
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java]: extra whitespace

File: impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java [hunk 12]
```diff
@@ -412,8 +412,8 @@ private boolean isDirectlyCacheable(Class<? extends Resource> clazz, Map<String,
 
         return AbstractResource.isMaterialized(data) &&
 
-               (!CollectionResource.class.isAssignableFrom(clazz) ||
-                (CollectionResource.class.isAssignableFrom(clazz) && isCollectionCachingEnabled()));
+                (!CollectionResource.class.isAssignableFrom(clazz) ||
+                        (CollectionResource.class.isAssignableFrom(clazz) && isCollectionCachingEnabled()));
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java]: extra whitespace

File: examples/spring-security-webmvc/src/main/java/com/stormpath/spring/examples/WebAppInitializer.java [hunk 13]
```diff
@@ -46,14 +46,15 @@ public void onStartup(ServletContext sc) throws ServletException {
         dispatcher.setLoadOnStartup(1);
         dispatcher.addMapping("/");
 
-        //Stormpath Filter
+        //Spring Security Filter: in front of Stormpath
+        FilterRegistration.Dynamic securityFilter = sc.addFilter(AbstractSecurityWebApplicationInitializer.DEFAULT_FILTER_NAME, DelegatingFilterProxy.class);
+        securityFilter.addMappingForUrlPatterns(EnumSet.allOf(DispatcherType.class), false, "/*");
+
+        //Stormpath Filter (after Spring Security)
         FilterRegistration.Dynamic filter = sc.addFilter("stormpathFilter", new DelegatingFilterProxy());
```
Comment (mrioan) [inline examples/spring-security-webmvc/src/main/java/com/stormpath/spring/examples/WebAppInitializer.java]: Sure, you are right. I will keep a consistent approach for both filters.

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/event/RequestEventListenerAdapter.java [hunk 14]
```diff
@@ -15,6 +15,7 @@
  */
 package com.stormpath.sdk.servlet.event;
 
+import com.stormpath.sdk.directory.CustomData;
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/event/RequestEventListenerAdapter.java]: Good catch. Removing now.

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractController.java [hunk 15]
```diff
@@ -165,6 +166,18 @@ public void setApplicationResolver(ApplicationResolver applicationResolver) {
         this.applicationResolver = applicationResolver;
     }
 
+    // Refactor of Provider requests for
+    // https://github.com/stormpath/stormpath-sdk-java/issues/915
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractController.java]: I do not recall exactly why now but the Social Login endpoints were also impacted by this. WRT to this method in particular this is not used at all. I will remove this method now.

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java (hunk 1 above)
Comment (mrioan) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java]: Done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java (hunk 3 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java]: Done

File: extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java (hunk 2 above)
Comment (mrioan) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java]: Done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java (hunk 4 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: Done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java (hunk 5 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: I did that on purpose, I think the text can be clearly read. Compare:

```
//Some  text here
```
vs
```
// Some text here
```

Agree? :)

I will add the space to the comment in line 287

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/LoginHandlerFilter.java (hunk 7 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/LoginHandlerFilter.java]: It was just a comment for myself which I did not see :) I will remove it now. Thanks.

Next time I will add a `TODO` in order to remind myself to read `TODO`s

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/security/provider/SocialCallbackSpringSecurityProcessingFilter.java [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+import java.io.IOException;
+
+/**
+ * A pre-authentication filter for Social accounts. Extracts an access token/code from the incoming callback request and
+ * uses it to populate the Spring Security context with an {@link ProviderAuthenticationToken ProviderAuthenticationToken}.
+ * <p>
+ *
+ * @since 1.3.0
+ */
+public class SocialCallbackSpringSecurityProcessingFilter extends HttpFilter implements InitializingBean {
```
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/security/provider/SocialCallbackSpringSecurityProcessingFilter.java]: @dogeared See? Spring Security deals with providers.

File: extensions/spring/stormpath-spring-security-webmvc/src/test/java/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcTestAppConfig.java (hunk 8 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/test/java/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcTestAppConfig.java]: The apply method is only removed in Spring Boot, this is a bare Spring test.

File: extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java (hunk 9 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java]: Surely a copy/paste and/or IDE change. I will improve this new

File: extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebMvcConfiguration.java (hunk 10 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebMvcConfiguration.java]: Done

File: impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java (hunk 11 above)
Comment (mrioan) [inline impl/src/main/java/com/stormpath/sdk/impl/ds/cache/WriteCacheFilter.java]: idem. Fixing now

**Comment by mrioan** [comment]
@dogeared, I see you approved the changes I made but I am not sure whether you consider this PR approved or not. Feel free to carry out another review. Thanks

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java [hunk 17]
```diff
@@ -114,15 +117,27 @@ public SamlResultListener springSecuritySamlResultListener() {
...
+    @Override
+    public SocialCallbackSpringSecurityProcessingFilter socialCallbackSpringSecurityProcessingFilter() {
+        return super.socialCallbackSpringSecurityProcessingFilter();
+    }
+
+    /**
+     * @since 1.3.0
+     */
+    @Bean
+    @ConditionalOnMissingBean(name="contentNegotiationSpringSecurityAuthenticationFilter")
```
Comment (dogeared) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java]: do we need the `name` param when the method is already the same?

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java [hunk 18]
```diff
@@ -162,17 +221,83 @@ public CsrfTokenManager stormpathCsrfTokenManager() {
...
+         */
+        filter.setProviderAccountRequestFactory(stormpathProviderAccountRequestFactory);
+
+        return filter;
     }
 
+    /**
+     * @since 1.3.0
+     */
+    public AccountResolverFilter springSecuritResolvedAccountFilter() {
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebSecurityConfiguration.java]: should be `springSecurityResolvedAccountFilter` (missing `y`)

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfiguration.java [hunk 19]
```diff
@@ -98,4 +111,28 @@ public AuthenticationEntryPoint stormpathAuthenticationEntryPoint() {
...
+    @Bean
+    public CorsConfigurationSource corsConfigurationSource() {
+        return super.corsConfigurationSource();
+    }
+
+    /**
+     * @since 1.3.0
+     */
+    @Bean
+    public AccountResolverFilter springSecuritResolvedAccountFilter() {
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfiguration.java]: add the `y` so it's `springSecurityResolvedAccountFilter`

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 20]
```diff
@@ -240,17 +276,27 @@ public void init(HttpSecurity http) throws Exception {
...
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: nit: should be `CORS` not `CORs`

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java [hunk 21]
```diff
@@ -35,19 +43,25 @@
...
     private List<MediaType> supportedMediaTypes;
+    private ProviderAccountRequestFactory providerAccountRequestFactory;
+
+    @Autowired
+    protected Application application;
 
     @Override
     public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
-        if (postOnly && !request.getMethod().equals("POST")) {
+        if (postOnly && !request.getMethod().equals("POST") && !request.getRequestURI().contains("/callbacks/")) {
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java]: Aren't the `callbacks` GETs? So, wouldn't the first clause cover this case?

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractSocialCallbackController.java [hunk 22]
```diff
@@ -55,7 +55,7 @@ public boolean isNotAllowedIfAuthenticated() {
         return true;
     }
 
-    protected abstract ProviderAccountRequest getAccountProviderRequest(HttpServletRequest request);
+    public abstract ProviderAccountRequest getAccountProviderRequest(HttpServletRequest request);
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractSocialCallbackController.java]: This method is invoked by `SocialCallbackSpringSecurityProcessingFilter which exists in another package`

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java [hunk 23]
```diff
@@ -132,7 +132,10 @@ protected ViewModel doGet(HttpServletRequest request, HttpServletResponse respon
     @SuppressWarnings("unchecked")
     protected Map<String,?> createModel(HttpServletRequest request, HttpServletResponse response) {
         List<ErrorModel> errors = null;
-        if (request.getParameter("error") != null) {
+        // We need to look for error in queryString now that we moved SpringSecurity to in in front of us: https://github.com/stormpath/stormpath-sdk-java/issues/915
+        String errorQueryString = request.getQueryString();
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java]: Makes sense

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java (hunk 17 above)
Comment (mrioan) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfiguration.java]: right, not needed. I will remove it.

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfiguration.java (hunk 19 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfiguration.java]: done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java (hunk 20 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: Done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java (hunk 21 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java]: -The first condition refers to an object instance flag (`postOnly`)
-The second condition is NOT POST (i.e GET)
-Callbacks are GET so a callback will succeed to pass first and second conditions.

Am I missing something?

**Comment by mrioan** [comment]
@dogeared I address your comments. Please note that there are 2 comments remaining. Back to you. Thanks!

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java (hunk 21 above)
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/ContentNegotiationSpringSecurityAuthenticationFilter.java]: oh, I see now. You *don't* want the `if` entered if it contains `/callbacks/`. I missed the `!`

**Comment by dogeared** [comment]
LGTM - please merge!

**Comment by dogeared** [comment]
@mrioan - I am going to merge so CI can pass.

**Pull request merged**